### PR TITLE
[MINOR] Eth peer toString() improvements

### LINKED ISSUE #6169: EthPeer.toString() method contains duplicated info
the full enode ID is in there 3 times - think this could be reduced to enhance readability

eg this string "10da28908de0fa66b3c01834576689a9d5f15c107c3f981c955391c81b9bbf04d489322b822d18aeb45ca629a99a3564c9987419c817d793790463388fd73064" appears below 3 times

extract from besu log:
`PeerId 0x10da28908de0fa66b3c01834576689a9d5f15c107c3f981c955391c81b9bbf04d489322b822d18aeb45ca629a99a3564c9987419c817d793790463388fd73064, reputation PeerReputation 101, validated? true, disconnected? false, client: Geth/v1.12.0-stable-e501b3b0/darwin-amd64/go1.20.3, connection [Connection with hashCode 1391819581 with peer 0x10da28908de0fa66b3c01834576689a9d5f15c107c3f981c955391c81b9bbf04d489322b822d18aeb45ca629a99a3564c9987419c817d793790463388fd73064 inboundInitiated false initAt 1700020096121], enode enode://10da28908de0fa66b3c01834576689a9d5f15c107c3f981c955391c81b9bbf04d489322b822d18aeb45ca629a99a3564c9987419c817d793790463388fd73064@127.0.0.1:30303`

### PR DESCRIPTION
**Pull request by macfarla**
* only include peer id in the peer.toString() method (remove from peerConnection)
* only log enode not id also
fixes #6169

### CODE REVIEW

**Comment by macfarla** [comment]
after this change
2023-11-16 13:42:11.027+10:00 | nioEventLoopGroup-3-1 | DEBUG | AbstractGetHeadersFromPeerTask | Received 1 of 1 headers requested from peer PeerId: 0x10da28908de0fa66b3c01834576689a9, PeerReputation 100, validated? true, disconnected? false, client: Geth/v1.12.0-stable-e501b3b0/darwin-amd64/go1.20.3, [Connection with hashCode -419500672 inboundInitiated? false initAt 1700106130632], enode://10da28908de0fa66b3c01834576689a9d5f15c107c3f981c955391c81b9bbf04d489322b822d18aeb45ca629a99a3564c9987419c817d793790463388fd73064@127.0.0.1:30303

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java [hunk 1]
```diff
@@ -593,8 +593,8 @@ public boolean hasSupportForMessage(final int messageCode) {
   @Override
   public String toString() {
     return String.format(
-        "PeerId %s, reputation %s, validated? %s, disconnected? %s, client: %s, connection %s, enode %s",
-        nodeId(),
+        "PeerId: %s, %s, validated? %s, disconnected? %s, client: %s, %s, %s",
+        getId().slice(0, 16),
```
Comment (siladu) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java:597]: If this is a truncated value maybe should indicate that, e.g. with an `...`?

**Review by siladu — COMMENTED** [review]
Good improvement 👍 

nit: Still quite a big log for DEBUG in my opinion. Would it make sense for this to be truncated further or moved to TRACE perhaps?
Maybe could even have a truncated one-liner DEBUG and a full TRACE version?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java (hunk 1 above)
Comment (macfarla) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java:597]: done

**Comment by macfarla** [comment]
revised 

```
2023-11-16 15:56:38.894+10:00 | nioEventLoopGroup-3-1 | TRACE | AbstractGetHeadersFromPeerTask | Updating chain state for peer 0xcbbaacaf47a005cb38 to block header 2017029 (0xf4738c1e95b017f372a69290b64540d87bc8559602ff18a436066e57ccf5
7279)
2023-11-16 15:56:38.896+10:00 | nioEventLoopGroup-3-1 | TRACE | AbstractGetHeadersFromPeerTask | Peer chain state ChainState{estimatedHeight=2017029, bestBlock=BestBlock{totalDifficulty=0x00000000000000000000000000000000000000000000000
000000000003d8e0b, blockHash=0xf4738c1e95b017f372a69290b64540d87bc8559602ff18a436066e57ccf57279, number=2017029}}
2023-11-16 15:56:38.896+10:00 | nioEventLoopGroup-3-1 | DEBUG | AbstractGetHeadersFromPeerTask | Received 1 of 1 headers requested from peer 0xcbbaacaf47a005cb38
```

**Review by siladu — APPROVED** [review]
Definite improvement.

nit: not consistently using the ellipsis - I'm hesitant to suggest putting it inside getShortNodeId, unless we made that explicitly indicate that method was for logging.

**Pull request merged**